app/models/v2/test_result.rb

```diff
@@ -63,6 +63,14 @@ class TestResult < ApplicationRecord
       }
     end
 
+    scope :with_groups, lambda { |groups, table = arel_table|
+      # Skip the [] representing ungrouped hosts from the array when generating the query
+      grouped = V2::System.arel_inventory_groups(groups.flatten, :id, table)
+      ungrouped = table[:groups].eq(AN::Quoted.new('[]'))
```

Comment:
Would ti be possible to have this special case (`AN::Quoted.new('[]')`) defined as a constant? Or is this the only place where it is used like that?